src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/MigrationReport.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Xproj
+{
+    /// <summary>
+    /// Migration report holder. Copied from
+    /// https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.ProjectJsonMigration/Report/MigrationReport.cs
+    /// </summary>
+    internal class MigrationReport
+    {
+        public List<ProjectMigrationReport> ProjectMigrationReports { get; set; }
+
+        public int MigratedProjectsCount { get; set; }
+
+        public int SucceededProjectsCount { get; set; }
+
+        public int FailedProjectsCount{ get; set; }
+
+        public bool AllSucceeded { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is MigrationReport report)
+            {
+                return Equals(ProjectMigrationReports, report.ProjectMigrationReports) &&
+                    Equals(MigratedProjectsCount, report.MigratedProjectsCount) &&
+                    Equals(SucceededProjectsCount, report.SucceededProjectsCount) &&
+                    Equals(FailedProjectsCount, report.FailedProjectsCount) &&
+                    Equals(AllSucceeded, report.AllSucceeded);
+            }
+            return false;
```

Comment:
It's for testing convenience. These structures are copied from the CLI (although the equals implementation I did myself), and we don't use them anywhere except here. I'll move the properties to private setters with `[JsonProperty]` attributes so it all works properly, and move the equals and hashcode methods to operate on the sequence instead of the list itself.